<!--
Coordinate Transform
X:  Forward
Y:  Left
Z:  Up
-->
<!--///Simulation of LIDAR///
    <node pkg="rosbag" type="play" name="play" args="-l /home/calibration/simulated_LIDARs_2018-06-11-10-05-58.bag"/>
-->

<launch>
    <param name="robot_description" textfile="$(find bsl_collision_avoidance)/urdf/CTA.urdf" />

    <!--///Static Transforms///-->
    <include file="$(dirname)/transforms.launch" />
    
    <!--///Laser Angle Listener for Calibration///-->
    <node pkg="bsl_collision_avoidance" type="calibration_listener.py" name="calibration_listener" output="screen" respawn="true"/>
    
    <node pkg="rosbag" type="play" name="play" args="-l /home/calibration/simulated_LIDARs_2018-06-11-10-05-58.bag"/>

    <!--///Running LIDAR 1 collision avoidance///-->
    <include file="$(dirname)/MRS6000_ns.launch">
        <arg name="name_space" value="1" /> 
        <arg name="laser_ip" value="192.168.1.7" />
        <arg name="laser_frame" value="laser1" />
        <arg name="parent_frame" value="left_lidar_mount" />
        <arg name="grid_frame" value="grid1" />
        <arg name="minimum_object_width" value="0.2" />
        <arg name="minimum_object_length" value="0.2" />
        <arg name="grid_height" value="4" />
        <arg name="grid_width" value="20" />
        <arg name="grid_x" value="5.5" />
        <arg name="grid_y" value="-2" />
        <arg name="modbus_address_output" value="17" />
        <arg name="modbus_address_input" value="0" />
        <arg name="zone_distance" value="5" />
    </include>
    
    <!--///Running LIDAR 2 collision avoidance///-->
    <include file="$(dirname)/MRS6000_ns.launch">
        <arg name="name_space" value="2" /> 
        <arg name="laser_ip" value="192.168.1.8" />
        <arg name="laser_frame" value="laser2" />
        <arg name="parent_frame" value="right_lidar_mount" />
        <arg name="grid_frame" value="grid2" />
        <arg name="minimum_object_width" value="0.2" />
        <arg name="minimum_object_length" value="0.2" />
        <arg name="grid_height" value="4" />
        <arg name="grid_width" value="20" />
        <arg name="grid_x" value="5.5" />
        <arg name="grid_y" value="-2" />
        <arg name="modbus_address_output" value="18"/>
        <arg name="modbus_address_input" value="0" />
        <arg name="zone_distance" value="5" />
    </include>
    
    <!--///Failure Output///-->
    <node pkg="bsl_collision_avoidance" type="failure_output.py" name="failure_output" respawn="true">
        <param name="modbus_address" value="16" />
    </node>
    
    <!--///System Shutoff on Digital Input 0 High///-->
<!--
    <node pkg="bsl_collision_avoidance" type="system_shutoff.py" name="system_shutoff" respawn="true" />
-->
    
</launch>
